### ISSUE TITLE
pyserial wrongly converted to Fedora package name

### ISSUE THREAD

**Issue by hroncok** [fedora-python/pyp2rpm#79]
```
python2 ./mybin.py -s esptool
```

Results in:

```
Requires:       python-pyserial
```

While the package name in Fedora for Python 2 is `pyserial`.

Tested on current master.

**Comment by mcyprian** [comment]
DantifiedNameConvertor wasn't used here, dnf can't be imported into python2. This is the result of NameConvertor's rpm_versioned_name method that simply adds ''python' prefix to the name of dependency.

Conversion works fine in DantifiedNameConvertor:

```
In [1]: from pyp2rpm.name_convertor import DandifiedNameConvertor
In [2]: con = DandifiedNameConvertor('fedora')
In [3]: con.rpm_versioned_name('pyserial', '2')
Out[3]: 'pyserial'
In [4]: con.rpm_versioned_name('pyserial', '3')
Out[4]: 'python3-pyserial'
```

**Comment by hroncok** [comment]
Why it cannot be imported?

**Comment by mcyprian** [comment]
It cannot be imported unless package python2-dnf is installed explicitly on Fedora. I added name_convertor info to log in 891ec8f8cc6553f501a9.

**Comment by hroncok** [comment]
I think a silent fial of import and falling back to not so good alternative is not a good idea. Do either of the following:
- Issue a warning if dnf cannot be imported with instructions about how to get it and explanation what's going to happen (i.e. the name conversion will not be perfect)
- Just drop the other name converter from the codebase and error if dnf cannot be imported.

Unless there is a good reason to support dnf-less environments, I'd say make ti hard dependency.

**Comment by mcyprian** [comment]
I added a warning in commit 891ec8f8cc6553f50. Is this satisfying solution @hroncok? I think there are good reasons to support dnf-less environments:
- 2 more distros - mageia and pld that don't contain dnf are supported by pyp2rpm
- We wouldn't be able to run many of the tests cases in Travis CI

**Comment by hroncok** [comment]
I created a testing sdist with syntax error in setup.py so I would test if the interpreters don't cycle there and back, which does not happen :+1: 

However, I found out that the error output is not nice on Python 3 as the primary interpreter:

```
$ python2 ./mybin.py -s ./*.gz
INFO  Pyp2rpm initialized.
INFO  Using /home/churchyard/rpmbuild/SOURCES as directory to save source.
INFO  Local file: ./isholiday-0.1.tar.gz copyed to /home/churchyard/rpmbuild/SOURCES/isholiday-0.1.tar.gz.
INFO  Getting metadata from setup.py using SetupPyMetadataExtractor.
INFO  Running extract_dist command using runpy.
ERROR  Error occured, trying alternative python interpreter.
ERROR  Subprocess failed, stdout , stderr:   File "setup.py", line 3
    )
    ^
SyntaxError: invalid syntax

Failed to extract data from setup.py script.
$ python3 ./mybin.py -s ./*.gz
INFO  Pyp2rpm initialized.
INFO  Using /home/churchyard/rpmbuild/SOURCES as directory to save source.
INFO  Local file: ./isholiday-0.1.tar.gz copyed to /home/churchyard/rpmbuild/SOURCES/isholiday-0.1.tar.gz.
INFO  Getting metadata from setup.py using SetupPyMetadataExtractor.
INFO  Running extract_dist command using runpy.
ERROR  Error occured, trying alternative python interpreter.
ERROR  Subprocess failed, stdout b'', stderr: b'  File "setup.py", line 3\n    )\n    ^\nSyntaxError: invalid syntax\n'
Failed to extract data from setup.py script.
```

Seems like some `bytes.decode()` call is missing.

 Attached the file for your convenience, it's public domain and rather small, so feel free to include it in the tests:
- [isholiday-0.1.tar.gz](https://github.com/fedora-python/pyp2rpm/files/482896/isholiday-0.1.tar.gz)

**Comment by hroncok** [comment]
I've also added a python 3 non compatible print statement now to see what happens:

```
$ python3 mybin.py -s ./isholiday-0.1.tar.gz
INFO  Pyp2rpm initialized.
INFO  Using /home/churchyard/rpmbuild/SOURCES as directory to save source.
INFO  Local file: ./isholiday-0.1.tar.gz copyed to /home/churchyard/rpmbuild/SOURCES/isholiday-0.1.tar.gz.
INFO  Getting metadata from setup.py using SetupPyMetadataExtractor.
INFO  Running extract_dist command using runpy.
ERROR  Error occured, trying alternative python interpreter.
ERROR  Failed to install package to virtualenv, skipping virtualenv metadata extraction.
WARNING  Some kind of error while communicating with client: None.
Traceback (most recent call last):
  File "/home/churchyard/Dokumenty/RedHat/pyp2rpm/pyp2rpm/metadata_extractors.py", line 39, in inner
    raise ValueError("Client is None.")
ValueError: Client is None.
WARNING  Template: fedora.spec was not found in /home/churchyard/Dokumenty/RedHat/pyp2rpm/fedora.spec using default template dir.
INFO  Using default template: fedora.spec.
INFO  Using name: python-isholiday.spec for specfile.
INFO  Specfile saved at: /home/churchyard/rpmbuild/SPECS/python-isholiday.spec.
INFO  That's all folks!
```

First of all, please notice the `ValueError: Client is None` part. I don't know why is that happening.

But the issue I wanted to point out here is the resulting spec file has python2 and python3 subpackages. I believe in a case when an interpreter fails to run setup.py, this version should be ruled out.
- [isholiday-0.1.tar.gz](https://github.com/fedora-python/pyp2rpm/files/482913/isholiday-0.1.tar.gz)

**Comment by mcyprian** [comment]
Thank you for your comments and testing data. 
- I corrected python3 byte decoding https://github.com/fedora-python/pyp2rpm/commit/620b4a8003bd15dd173da3870230cf94207d3512#diff-3509bb8546c144c930870f6ffd947b2cR61
- Currently running version of python is removed from versions after runpy failure 849945a189416f7
- Client is disabled if input of pyp2rpm is a local file. It is true that warning in example above doesn't make sense. I tried to improve warning message in fcffbc75aaefd. I am not sure if we want to run pyp2rpm in "offline mode" when working with local files like it is now or try to create the client anyway.

**Comment by hroncok** [comment]
BTW My comments were initially meant for the PR, sorry about that.

**Comment by glensc** [comment]
btw, i think it's best to add `Requires` to fedora version of `pyp2rpm` package that is included in distro. pyp2rpm is kind of development tool, having it python-dnf installed is good choice there.

**Comment by mcyprian** [comment]
Fixed in 891ec8f8cc6553f.
